src/main/java/core/basesyntax/supplier/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    public static Figure getRandomFigure() {
+        Random random = new Random();
+        switch (random.nextInt(4) + 1) {
+            case 1:
+                return new Circle(ColorSupplier.getRandomColor(), random.nextInt(8) + 1);
+            case 2:
+                return new Square(ColorSupplier.getRandomColor(), random.nextInt(8) + 1);
+            case 3:
+                return new Rectangle(ColorSupplier.getRandomColor(), random.nextInt(8) + 1,
+                        random.nextInt(8) + 1);
+            case 4:
+                return new RightTriangle(ColorSupplier.getRandomColor(), random.nextInt(8) + 1,
+                        random.nextInt(8) + 1);
+            case 5:
+                return new IsoscelesTrapezoid(ColorSupplier.getRandomColor(), random.nextInt(8) + 1,
+                        random.nextInt(8) + 1);
+            default:
+                return getDefaultFigure();
+        }
```

Comment:
lets make 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction. Also lets moe magic numbers 4, and 8 to constants https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names

Reply:
Why i need creat a 5 methods? What`s wrong with my solution?